bundles/org.eclipse.swt/Eclipse SWT Browser/win32/org/eclipse/swt/browser/Edge.java

```diff
@@ -282,6 +279,104 @@ static int callAndWait(String[] pstr, ToIntFunction<IUnknown> callable) {
 	return phr[0];
 }
 
+class WebViewProvider {
+
+	private CompletableFuture<ICoreWebView2> webViewFuture = new CompletableFuture<>();
+	private CompletableFuture<ICoreWebView2_2> webView_2Future = new CompletableFuture<>();
+	private CompletableFuture<ICoreWebView2_11> webView_11Future = new CompletableFuture<>();
+
+	private CompletableFuture<Void> lastWebViewTask = getWaitForWebViewInitializationFuture();
+
+
+	private CompletableFuture<Void> getWaitForWebViewInitializationFuture() {
+		return webViewFuture.thenCompose(__ -> {
+			return CompletableFuture.allOf(webView_2Future.handle((result, ex) -> null), webView_11Future.handle((result, ex) -> null));
+		});
+	}
+
+	void initializeWebView(ICoreWebView2Controller controller) {
+		long[] ppv = new long[1];
+		controller.get_CoreWebView2(ppv);
+		webViewFuture.complete(new ICoreWebView2(ppv[0]));
+		initializeWebView_2();
+		initializeWebView_11();
```

Comment:
nitpicky: since you already have the web biew instance at hand here (created in the line before), you could directly pass it to these methods instead of performing another query of the web view within those methods.

Reply:
Yeah right

Reply:
I could also return the webView from here so they dont have to call getWebView later